### PR TITLE
feat(language): parameter for version detection

### PR DESCRIPTION
**Pull request by lnu**
### Prerequisites

- [x] I have read and understood the `CONTRIBUTING` guide
- [x] The commit message follows the [conventional commits][cc] guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes/features)

read angular version from angular project file. New parameter added for language  for custom version reading.

[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

### CODE REVIEW

File: src/segment_language.go [hunk 1]
```diff
@@ -166,13 +168,22 @@ func (l *language) hasLanguageFiles() bool {
...
+		if command.getVersion == nil {
+			if !l.env.hasCommand(command.executable) {
+				continue
+			}
+			versionStr, err = l.env.runCommand(command.executable, command.args...)
+			if exitErr, ok := err.(*commandError); ok {
+				l.exitCode = exitErr.exitCode
+				return fmt.Errorf("err executing %s with %s", command.executable, command.args)
+			}
+		} else {
```
Comment (JanDeDobbeleer) [inline src/segment_language.go:182]: if we swap the if, we can simply return early or continue in case we have an error

**Review by JanDeDobbeleer — CHANGES_REQUESTED** [review]
One remark

**Comment by JanDeDobbeleer** [comment]
@lnu question, do we ever see the need for additional logic to add that to cmd but rather have the function live on language level and only execute it once?

**Comment by lnu** [comment]
> @lnu question, do we ever see the need for additional logic to add that to cmd but rather have the function live on language level and only execute it once?

I put it at the command level because we still use the regex parsing part of the command.

**Comment by JanDeDobbeleer** [comment]
@lnu fair point, can you have a look at my comment? Then we can merge this in.

File: src/segment_language.go (hunk 1 above)
Comment (lnu) [reply src/segment_language.go:182]: I don't really get it. If `getVersion` is not null we use it to read the version, if not we use the command. Both will output a `versionStr` that needs to be parsed using the regex. I probably missed your point, sorry.
image

Comment (JanDeDobbeleer) [reply src/segment_language.go:182]: you're right.

**Pull request merged**